stores/directory/local.py

```diff
@@ -23,42 +26,81 @@ def __init__(self, local_dir: Union[str, Path]):
         assert local_dir_path.is_dir(), f"Given path {local_dir_path} is not a directory."
 
         # Store that location against the class
-
         self.local_path = local_dir_path
 
-
     def add_file(self, file: Path):
         # Abstract methods can remain without implementation for now.
         pass
 
     def has_lone_file_matching(self, pattern: str) -> bool:
-        #  To be implemented in local directory store 1 #5 
-        ...
-
-    def save_lone_file_matching(self, pattern: str, save_as: str) -> Path:
-        #  To be implemented in local directory store 1 #5 
-        ...
+        # Grab a list of files matching the regex pattern to determine how many exist.
+        matching_files = [
+            f for f in self.get_file_names() if re.search(pattern, f)]
+
+        if len(matching_files) == 1:  # 1 file matched
+            return True
+        elif len(matching_files) == 0:  # 0 file matched
+            raise FileNotFoundError(f"No files found matching given pattern '{pattern}' in directory {self.local_path}")          
+        elif len(matching_files) > 1:  # 2+ files matched
+            raise FileNotFoundError(f"More than 1 file found that matches the regex pattern '{pattern}' in directory {self.local_path}")
+
+    def save_lone_file_matching(self, pattern: str, destination: Optional[Union[Path, str]] = None):
+        # Assert 1 file matches
+        if self.has_lone_file_matching(pattern):
+            file_name_list = self.get_file_names()
+            file_name = file_name_list[0]  
+            
+            if not destination:
+                open(file_name, "x")
+            else:
+                if isinstance(destination, str):
+                    save_path = Path(destination)
+                    save_path = save_path / file_name
+                    assert not save_path.exists(), f"Given file already exists in directory {save_path}."
+                    with (save_path).open("w") as saved_pattern:
+                        pass
+                else:
+                    save_path = destination / file_name
+                    assert not save_path.exists(), f"Given file already exists in directory {save_path}."
+                    with (save_path).open("w") as saved_pattern:
+                        pass
 
     def get_lone_matching_json_as_dict(self, pattern: str) -> dict:
-        #  To be implemented in local directory store 1 #5 
-        ...
+        # Assert 1 file matches
+        if self.has_lone_file_matching(pattern): 
+            file_names = self.get_file_names()
```

Comment:
double check this, but I think because the conditional == True, you can just use `Path(self._files_that_match_pattern(pattern)[0])` to immediately get the Path to the file you want.

then just skip straight to whats currently line 79